src/Simplex/Chat.hs

```diff
@@ -2135,31 +2136,43 @@ processChatCommand' vr = \case
-          contacts <-
-            filter (\ct -> contactReady ct && contactActive ct && not (contactConnIncognito ct))
-              <$> withStore' (`getUserContacts` user)
+          contacts <- withStore' (`getUserContacts` user)
           user' <- updateUser
           asks currentUser >>= atomically . (`writeTVar` Just user')
           withChatLock "updateProfile" . procCmd $ do
-            ChatConfig {logLevel} <- asks config
-            summary <- foldM (processAndCount user' logLevel) (UserProfileUpdateSummary 0 0 0 []) contacts
+            let (notChanged, changedCts) = foldr (addChangedProfileContact user') (0, []) contacts
+                idsEvts = map ctSndMsg changedCts
+            msgReqs_ <- zipWith ctMsgReq changedCts <$> createSndMessages idsEvts
+            (errs, cts) <- partitionEithers . zipWith (second . const) changedCts <$> deliverMessagesB msgReqs_
+            unless (null errs) $ toView $ CRChatErrors (Just user) errs
+            let changedCts' = filter (\ChangedProfileContact {ct, ct'} -> directOrUsed ct' && mergedPreferences ct' /= mergedPreferences ct) cts
+            createContactsSndFeatureItems user' changedCts'
+            let summary =
+                  UserProfileUpdateSummary
+                    { notChanged,
+                      updateSuccesses = length cts,
+                      updateFailures = length errs,
+                      changedContacts = map (\ChangedProfileContact {ct'} -> ct') changedCts'
+                    }
             pure $ CRUserProfileUpdated user' (fromLocalProfile p) p' summary
       where
-        processAndCount user' ll s@UserProfileUpdateSummary {notChanged, updateSuccesses, updateFailures, changedContacts = cts} ct = do
-          let mergedProfile = userProfileToSend user Nothing $ Just ct
-              ct' = updateMergedPreferences user' ct
-              mergedProfile' = userProfileToSend user' Nothing $ Just ct'
-          if mergedProfile' == mergedProfile
-            then pure s {notChanged = notChanged + 1}
-            else
-              let cts' = if mergedPreferences ct == mergedPreferences ct' then cts else ct' : cts
-               in (notifyContact mergedProfile' ct' $> s {updateSuccesses = updateSuccesses + 1, changedContacts = cts'})
-                    `catchChatError` \e -> when (ll <= CLLInfo) (toView $ CRChatError (Just user) e) $> s {updateFailures = updateFailures + 1, changedContacts = cts'}
+        -- [incognito] filter out contacts with whom user has incognito connections
+        addChangedProfileContact :: User -> Contact -> (Int, [ChangedProfileContact]) -> (Int, [ChangedProfileContact])
+        addChangedProfileContact user' ct (notChangedCnt, changedCts) = case contactSendConn_ ct' of
+          Left _ -> (notChangedCnt, changedCts)
+          Right conn
+            | connIncognito conn -> (notChangedCnt, changedCts)
+            | mergedProfile' == mergedProfile -> (notChangedCnt + 1, changedCts)
```

Comment:
```suggestion
            | connIncognito conn || mergedProfile' == mergedProfile -> (notChangedCnt + 1, changedCts)
```

Reply:
incognito contacts weren't included into this count previously - here they're treated same as contacts without connections / with inactive connections - they shouldn't ever receive profile update, and only non incognito contacts with active connections should so they are counted if it's not changed

Reply:
not sure it makes sense tbh. They are not changed, right?

Reply:
I think it's more logical to count them as unchanged than to have them as "unaccounted"